scripts/valhalla_build_config

```diff
@@ -577,9 +566,19 @@ def add_leaf_args(path, tree, leaves, parser):
     h = help_text
     for k in keys:
       h = h[k]
-    t = comma_separated_list if type(tree) == type([]) else (type(tree) if type(tree) != type(optional()) else tree.type)
+
+    # its either required and is the right type or optional and has a type to use
+    # lists are supported as comma separated and bools support a bunch of conventions
+    # the rest of the primatives (strings and numbers) parse automatically
+    t = type(tree) if type(tree) != type(optional()) else tree.type
```

Comment:
this is the real trick here. previously optional was only supported for string and number types. now we first resolve optional to the type and then we add the special parsing functions. this allows `optional(list)` to be su pported now